algorithms/src/Kokkos_Sort.hpp

```diff
@@ -628,7 +624,31 @@ std::enable_if_t<Kokkos::is_execution_space<ExecutionSpace>::value> sort(
   bin_sort.sort(exec, view);
 }
 
+template <class ExecutionSpace, class ViewType>
+KOKKOS_DEPRECATED_WITH_COMMENT(
+    "Use the overload not taking bool always_use_kokkos_sort")
+std::enable_if_t<Kokkos::is_execution_space<ExecutionSpace>::value> sort(
+    const ExecutionSpace& exec, ViewType const& view,
+    bool const always_use_kokkos_sort) {
+  if (!always_use_kokkos_sort && Impl::try_std_sort(view, exec)) {
+    return;
+  } else {
+    sort(exec, view);
+  }
+}
+
+template <class ViewType>
+void sort(ViewType const& view) {
+  typename ViewType::execution_space exec;
+  // FIXME: Do this and the following deprecated overload need to call
+  // Kokkos::fence() first?
+  sort(exec, view);
+  exec.fence("Kokkos::Sort: fence after sorting");
+}
```

Comment:
I think it's OK to fix/change that later in `develop` (where I already had the same comment).